### PR TITLE
fix(gatsby): Hot reload package and theme queries

### PR DESCRIPTION
**Pull request by sidharthachatterjee**
Fixes https://github.com/gatsbyjs/gatsby/issues/15726

- [x] Add a failing test 
- [x] Make the fix

### CODE REVIEW

File: packages/gatsby/src/query/query-watcher.js [hunk 1]
```diff
@@ -220,11 +221,22 @@ const debounceCompile = _.debounce(() => {
...
-const watch = rootDir => {
+const watch = async rootDir => {
   if (watcher) return
 
+  const modulesThatUseGatsby = await getGatsbyDependents()
+
+  const packagePaths = modulesThatUseGatsby.map(module => {
+    const filesRegex = `*.+(t|j)s?(x)`
+    const pathRegex = `/{${filesRegex},!(node_modules)/**/${filesRegex}}`
+    slash(path.join(module.path, pathRegex))
```
Comment (pieh) [inline packages/gatsby/src/query/query-watcher.js]: Shouldn't this be:
```suggestion
    return slash(path.join(module.path, pathRegex))
```
?

Comment (sidharthachatterjee) [reply packages/gatsby/src/query/query-watcher.js]: Oops

**Comment by sidharthachatterjee** [comment]
@pieh This is good to go now! I've skipped the test for now

**Comment by sidharthachatterjee** [comment]
Published in `gatsby@2.13.27`

**Pull request merged**